src/SprykerEco/Yves/Payone/Theme/default/components/molecules/payone-klarna/payone-klarna.ts

```diff
@@ -0,0 +1,162 @@
+/* tslint:disable: no-any */
+/* tslint:disable: max-file-line-count */
+declare var Klarna: any;
+
+import Component from 'ShopUi/models/component';
+import ScriptLoader from 'ShopUi/components/molecules/script-loader/script-loader';
+
+const TOKEN_CONTAINER_ID = 'paymentForm_payoneKlarna_payMethodToken';
+const IS_VALID_PARAM = 'is_valid';
+const GET_TOKEN_URL = '/payone/get-token';
+const CONTAINER_ID = '#klarna_container';
+
+export default class PayoneKlarna extends Component {
+    protected scriptLoader: ScriptLoader;
+    protected selectField: HTMLSelectElement;
+    protected availablePayment = {};
+    protected availablePaymentArray = [];
+    protected currentPaymentMethodCategory: string;
+    protected currentPaymentCompanyToken: string;
+    protected allKlarnaPayMethods: [];
+
+    protected readyCallback() {}
+
+    protected init() {
+        this.scriptLoader = <ScriptLoader>this.querySelector('script-loader');
+        this.selectField = <HTMLSelectElement>this.getElementsByClassName(this.selectFieldClassName)[0];
+        this.allKlarnaPayMethods = JSON.parse(this.klarnaPayMethods());
+
+        this.mapEvents();
+    }
+
+    protected mapEvents(): void {
+        this.scriptLoader.addEventListener('scriptload', (event: Event) => this.onScriptLoad(event));
+        this.selectField.addEventListener('change', (event: Event) => this.selectPayMethod(event));
+    }
+
+    protected onScriptLoad(event: Event): void {
+        this.getAvailableMethods();
+    }
+
+    protected getAvailableMethods(): void {
+        Array.from(this.selectField.options).map((option) => {
+            if (!option.value) {
+                return;
+            }
+
+            const formData = new FormData();
+            formData.append('pay_method', option.value);
+
+            fetch(GET_TOKEN_URL, {method: 'POST', body: formData})
+                .then((response) => response.json())
+                .then((parsedResponse) => {
+                    if (parsedResponse[IS_VALID_PARAM]) {
+                        this.availablePayment = {
+                            'pay_method': option.value,
+                            'client_token': parsedResponse['client_token'],
+                        };
+
+                        this.availablePaymentArray.push(this.availablePayment);
+
+                        option.removeAttribute('disabled');
+                        return;
+                    }
+
+                })
+                .catch((e: Error) => {
+                    console.log(e.message);
+                });
+        })
+    }
+
+    protected selectPayMethod(event: Event): void {
+        const currentParams = this.availablePaymentArray.filter(payment => {
+            return payment['pay_method'] === this.selectField.value;
+        })
+
+        this.loadKlarna(currentParams[0])
+    }
```

Comment:
`event` parameter is not used.
Also if is used the first element it's better to use `find()` instead of `filter()`.
```suggestion
    protected selectPayMethod(): void {
        const currentParams = this.availablePaymentArray.filter(payment => payment['pay_method'] === this.selectField.value);

        this.loadKlarna(currentParams[0]);
    }
```